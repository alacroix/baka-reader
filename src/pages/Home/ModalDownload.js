// @flow

import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Modal from 'react-native-modal';
import appStyle from 'bakareader/src/appStyle';
import { Button, Input } from 'bakareader/src/components';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'stretch',
    marginHorizontal: '15%',
  },
  modal: {
    backgroundColor: 'white',
    borderRadius: 4,
    borderColor: 'rgba(0, 0, 0, 0.1)',
  },
  modalBody: {
    padding: appStyle.grid.x3,
  },
  modalHeader: {
    marginBottom: 20,
    fontSize: appStyle.font.size.huge,
  },
  modalContent: {},
  label: {
    marginBottom: appStyle.grid.x1,
  },
  modalFooter: {
    justifyContent: 'flex-end',
    flexDirection: 'row',
  },
});

type StateType = {
  url: string,
}

type PropsType = {
  isVisible: boolean,
  onClose: () => void,
  onConfirm?: ({url: string}) => void,
}

class ModalDownload extends Component {
  static defaultProps = {
    onConfirm: () => {},
  }

  state: StateType;

  componentWillMount() {
    this.onConfirm = this.onConfirm.bind(this);
    this.onInputChange = this.onInputChange.bind(this);
    this.setState({
      url: '',
    });
  }

  onConfirm: Function;
  onInputChange: Function;

  onConfirm() {
    if (this.props.onConfirm) {
      this.props.onConfirm({ url: this.state.url });
      this.props.onClose();
    }
  }

  onInputChange(newUrl: string) {
    this.setState({
      url: newUrl,
    });
  }

  props: PropsType;

  render() {
    return (
      <Modal isVisible={this.props.isVisible} style={styles.container}>
        <View style={styles.modal}>
          <View style={styles.modalBody}>
            <Text style={styles.modalHeader}>Ajouter un livre</Text>
            <View style={styles.modalContent}>
              <Text style={styles.label}>URL</Text>
              <Input
                onChangeText={this.onInputChange}
                placeholder="URL du livre"
                type="url"
                value={this.state.url}
              />
            </View>
          </View>
          <View style={styles.modalFooter}>
            <Button onPress={this.props.onClose} type="flat">Annuler</Button>
            <Button onPress={this.onConfirm} type="flat">Télécharger</Button>
          </View>
        </View>
      </Modal>
    );
  }
}

export default ModalDownload;
