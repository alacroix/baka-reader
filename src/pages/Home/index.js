// @flow

import React, { Component } from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';

import { Button, Grid, Page } from 'bakareader/src/components';
import { getBookInfos } from 'bakareader/src/services/FileSystem';
import appStyle from 'bakareader/src/appStyle';

import ModalDownload from './ModalDownload';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  header: {
    fontSize: appStyle.font.size.huge,
    marginTop: appStyle.grid.x2,
    marginLeft: appStyle.grid.x2,
  },
});

type PropsType = {
  books: Array<any>,
  navigation: any,
};

class Home extends Component {
  static navigationOptions = {
    title: 'Home',
    headerLeft: null,
    gesturesEnabled: false,
  };

  state = {
    modalVisible: false,
  }

  onBookPress(book: RNFetchBlobStat) {
    getBookInfos(book.path)
      .then(infos => this.props.navigation.navigate('infos', { book, currentPage: 1, infos }));
  }

  setModalVisible(visible: boolean) {
    this.setState({ modalVisible: visible });
  }

  props: PropsType;

  render() {
    const books = this.props.navigation.state.params.books;
    return (
      <Page noMargin>
        <View style={styles.container}>
          <ScrollView showsVerticalScrollIndicator={false}>
            <Text style={styles.header}>
              Collection
            </Text>
            <Grid books={books} onBookPress={book => this.onBookPress(book)} />
            <Button onPress={() => { this.setModalVisible(true); }}>Go to the Download modal</Button>
            <ModalDownload
              isVisible={this.state.modalVisible}
              onClose={() => this.setModalVisible(false)}
            />
          </ScrollView>
        </View>
      </Page>
    );
  }
}

export default Home;
