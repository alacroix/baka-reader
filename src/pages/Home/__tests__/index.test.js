import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Home from '../index';

jest.mock('../../../services/BookManager');

describe('Home', () => {
  it('should render itself', () => {
    const navigation = {
      state: {
        params: {
          books: [],
        },
      },
    };

    const tree = renderer.create(
      <Home navigation={navigation} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
