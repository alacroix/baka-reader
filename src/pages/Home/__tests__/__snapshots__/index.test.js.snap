// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render itself 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      Object {
        "backgroundColor": undefined,
        "paddingHorizontal": 0,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      showsVerticalScrollIndicator={false}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "marginLeft": 16,
              "marginTop": 16,
            }
          }
        >
          En cours
        </Text>
        <View
          style={
            Object {
              "marginBottom": 16,
              "marginTop": 32,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            horizontal={true}
            showsHorizontalScrollIndicator={false}
          >
            <View />
          </RCTScrollView>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "marginLeft": 16,
              "marginTop": 16,
            }
          }
        >
          Collection
        </Text>
        <View
          style={
            Object {
              "marginTop": 16,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "paddingBottom": 32,
                "paddingTop": 16,
              }
            }
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={4}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            onViewableItemsChanged={undefined}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsVerticalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "minHeight": 48,
              "minWidth": 64,
              "opacity": 1,
              "paddingHorizontal": 8,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "height": 36,
                  "justifyContent": "center",
                  "paddingHorizontal": 8,
                },
                Object {
                  "backgroundColor": "#05A5D1",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#FAFAFA",
                  },
                ]
              }
            >
              GO TO THE DOWNLOAD MODAL
            </Text>
          </View>
        </View>
        <Modal
          animationType="none"
          hardwareAccelerated={false}
          hideOnBack={true}
          onBackButtonPress={[Function]}
          onRequestClose={[Function]}
          transparent={true}
          visible={false}
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "black",
                "bottom": 0,
                "height": 1334,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 750,
              }
            }
          />
          <View
            collapsable={undefined}
            hideOnBack={true}
            onBackButtonPress={[Function]}
            style={
              Object {
                "alignItems": "stretch",
                "flex": 1,
                "justifyContent": "center",
                "margin": 37.5,
                "marginHorizontal": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "rgba(0, 0, 0, 0.1)",
                  "borderRadius": 4,
                }
              }
            >
              <View
                style={
                  Object {
                    "padding": 24,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                      "marginBottom": 20,
                    }
                  }
                >
                  Ajouter un livre
                </Text>
                <View
                  style={Object {}}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "marginBottom": 8,
                      }
                    }
                  >
                    URL
                  </Text>
                  <TextInput
                    autoCapitalize="none"
                    autoCorrect={false}
                    keyboardType="url"
                    onChangeText={[Function]}
                    placeholder="URL du livre"
                    style={
                      Object {
                        "borderColor": "gray",
                        "borderWidth": 1,
                        "height": 40,
                        "paddingHorizontal": 8,
                      }
                    }
                    type="url"
                    value=""
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "minHeight": 48,
                      "minWidth": 64,
                      "opacity": 1,
                      "paddingHorizontal": 8,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 36,
                          "justifyContent": "center",
                          "paddingHorizontal": 8,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "fontSize": 13,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "#05A5D1",
                          },
                        ]
                      }
                    >
                      ANNULER
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "minHeight": 48,
                      "minWidth": 64,
                      "opacity": 1,
                      "paddingHorizontal": 8,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 36,
                          "justifyContent": "center",
                          "paddingHorizontal": 8,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "fontSize": 13,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "#05A5D1",
                          },
                        ]
                      }
                    >
                      TÉLÉCHARGER
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
